### PR TITLE
Test/stabilization5

### PR DESCRIPTION
**Pull request by clarissacastella**
Please review.

### CODE REVIEW

File: test/spec/app_spec.rb [hunk 1]
```diff
@@ -585,28 +580,34 @@
...
       @driver.action.send_keys(:down).perform
       @driver.action.send_keys(:return).perform
-      sleep 2
+      sleep 1
       @driver.navigate.refresh
-      sleep 3
+      sleep 2
+      wait_for_size_change(0, "sourceLanguageInput",:id)
+      wait_for_selector("source__tab-button-media",:class)      
       expect(@driver.page_source.include?('Acoli')).to be(true)
```
Comment (caiosba) [inline test/spec/app_spec.rb]: Shouldn't this be asserted before the `@driver.navigate.refresh` in order to be sure that auto-refresh worked?

File: test/spec/app_spec.rb [hunk 2]
```diff
@@ -585,28 +580,34 @@
...
-      list = @driver.find_elements(:css => "div.source-tags__tag svg")
-      list[0].click
+      elements =wait_for_selector_list("div.source-tags__tag svg")
+      elements[0].click
       sleep 1
       @driver.navigate.refresh
-      sleep 3
+      sleep 2
+      wait_for_selector("source__tab-button-media",:class)
       expect(@driver.page_source.include?('Acoli')).to be(false)
```
Comment (caiosba) [inline test/spec/app_spec.rb]: Same as above. Assert that the language was removed even if the page was not refreshed.

File: test/spec/app_spec.rb [hunk 3]
```diff
@@ -1038,11 +1018,32 @@
       expect(@driver.page_source.include?('Auto-Refresh')).to be(true)
     end
 
-    # it "should cancel request through switch teams" do
-    #   skip("Needs to be implemented")
-    # end
+    it "should give 404 when trying to acess a media that is not related to the project on the URL", bin1: true do
```
Comment (caiosba) [inline test/spec/app_spec.rb]: Typo: `acess` -> `access`

File: test/spec/app_spec.rb [hunk 4]
```diff
@@ -1038,11 +1018,32 @@
...
+      el.click
+
+      fill_field('team-name-container', team, :id)
+      el = wait_for_selector('team-slug-container',:id)
+      el.click
+      el.clear
+      fill_field('team-slug-container', team, :id)
+      el = wait_for_selector("create-team__submit-button",:class)
+      el.click
+      wait_for_selector("team-members__profile-link",:class)
```
Comment (caiosba) [inline test/spec/app_spec.rb]: From line 1023 until here: why do you need to create a team using the web UI?

File: test/spec/app_spec.rb [hunk 5]
```diff
@@ -1038,11 +1018,32 @@
...
+      url = @config['self_url'] + '/'+ team +'/project/'+data.project_id.to_s + '/media/'+data.media_id.to_s
```
Comment (caiosba) [inline test/spec/app_spec.rb]: You tested something different: you tested that the media is not related to the team, not the project. I suggest that you just call `api_create_team_project_and_claim_and_redirect_to_media_page` (which is faster than a link and you don't need necessarily a link here), get the current URL, replace the project id in the URL by `0` and assert that you reach a "Not Found" page.

File: test/spec/app_spec.rb [hunk 6]
```diff
@@ -1290,7 +1344,7 @@
         expect(@driver.page_source.include?('Embed')).to be(false)
         @driver.navigate.to "#{@driver.current_url}/embed"
         sleep 2
-        expect(@driver.page_source.include?('Not available')).to be(true)
+        expect(@drivre.page_source.include?('Not available')).to be(true)
```
Comment (caiosba) [inline test/spec/app_spec.rb]: Should be `driver`

File: test/spec/app_spec.rb [hunk 7]
```diff
@@ -1137,9 +1138,62 @@
...
+      el = wait_for_selector('.task-actions__edit-response', :css)
+      el.click
+      el = wait_for_selector('1', :id)
+      el.click
+      el = wait_for_selector('task__submit', :class)
+      el.click
+      media_pg.wait_all_elements(8, "annotations__list-item", :class) #Wait for refresh page
+      expect(@driver.page_source.gsub(/<\/?[^>]*>/, '').include?('Task "Foo or bar???" answered by User With Email: "Bar"')).to be(true)
+      # Delete task
+      delete_task('Foo')
```
Comment (caiosba) [inline test/spec/app_spec.rb]: Does this method assert that the task was actually deleted?

File: test/spec/app_spec.rb (hunk 1 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: I think the result is the same, `wait_for_size_change(0, "sourceLanguageInput",:id)` being before or after the refresh. But I'll change.

The `wait_for_selector("source__tab-button-media",:class) ` needs to be after, it ensures that the page is completely loaded.

File: test/spec/app_spec.rb (hunk 2 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: The `wait_for_selector("source__tab-button-media",:class) ` needs to be after refresh, it ensures that the page is completely loaded.

File: test/spec/app_spec.rb (hunk 3 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: :+1:

File: test/spec/app_spec.rb (hunk 1 above)
Comment (caiosba) [reply test/spec/app_spec.rb]: I mean the line `expect(@driver.page_source.include?('Acoli')).to be(true)`. You're checking after the `@driver.navigate.refresh`. We need to be sure that the language is added after the user presses `ENTER`. The user shouldn't need to refresh the whole page to see the new language he/she added.

File: test/spec/app_spec.rb (hunk 2 above)
Comment (caiosba) [reply test/spec/app_spec.rb]: Same as https://github.com/meedan/check-web/pull/217#discussion_r151712252

File: test/spec/app_spec.rb (hunk 4 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: It is not necessary.
Updated to `      api_create_team(team: team)`

Comment (clarissacastella) [reply test/spec/app_spec.rb]: Updated to `api_create_team_and_project()` indeed

File: test/spec/app_spec.rb (hunk 5 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: Updated to use  `api_create_team_and_project()` + `api_create_team_project_and_link` to use a real project id instead of `0`

File: test/spec/app_spec.rb (hunk 6 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: :+1:

File: test/spec/app_spec.rb (hunk 7 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: I think it so.
https://github.com/meedan/check-web/blob/develop/test/spec/app_spec_helpers.rb#L81

Comment (clarissacastella) [reply test/spec/app_spec.rb]: I think it so
https://github.com/meedan/check-web/blob/develop/test/spec/app_spec_helpers.rb#L81

Comment (caiosba) [reply test/spec/app_spec.rb]: Right!

File: test/spec/app_spec.rb [hunk 8]
```diff
@@ -585,28 +580,34 @@
...
       @driver.action.send_keys(:down).perform
       @driver.action.send_keys(:return).perform
-      sleep 2
+      sleep 1
+      wait_for_size_change(0, "sourceLanguageInput",:id)
       @driver.navigate.refresh
-      sleep 3
+      sleep 2
+      wait_for_selector("source-tags__tag",:class)  
       expect(@driver.page_source.include?('Acoli')).to be(true)
```
Comment (caiosba) [inline test/spec/app_spec.rb]: Shouldn't we copy (not move) the line `expect(@driver.page_source.include?('Acoli')).to be(true)` to after line 597, before the `@driver.navigate.refresh`?

File: test/spec/app_spec.rb [hunk 9]
```diff
@@ -585,28 +580,34 @@
...
-      list = @driver.find_elements(:css => "div.source-tags__tag svg")
-      list[0].click
+      elements =wait_for_selector_list("div.source-tags__tag svg")
+      elements[0].click
       sleep 1
       @driver.navigate.refresh
-      sleep 3
+      sleep 2
+      wait_for_selector("source__tab-button-media",:class)
       expect(@driver.page_source.include?('Acoli')).to be(false)
```
Comment (caiosba) [inline test/spec/app_spec.rb]: Same as above... shouldn't we copy `expect(@driver.page_source.include?('Acoli')).to be(false)` to after line 607 (before `@driver.navigate.refresh`) in order to be sure that the language was removed without needing to refresh the page?

File: test/spec/app_spec.rb (hunk 8 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: The idea is to see if the tag with the language is being viewed in the page not in the edit mode.

File: test/spec/app_spec.rb (hunk 9 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: The idea is to see if the tag with the language was deleted in the page not in the edit mode.

File: test/spec/app_spec.rb (hunk 8 above)
Comment (caiosba) [reply test/spec/app_spec.rb]: But after we press "Enter" we exit edit mode. I mean that we don't need to refresh the page to see that the item was actually saved. We can just press "Save". Here: https://www.youtube.com/watch?v=a7w4qDacORA

File: test/spec/app_spec.rb (hunk 9 above)
Comment (caiosba) [reply test/spec/app_spec.rb]: But after we press "Enter" we exit edit mode. I mean that we don't need to refresh the page to see that the item was actually saved. We can just press "Save". Here: https://www.youtube.com/watch?v=a7w4qDacORA

File: test/spec/app_spec.rb (hunk 8 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: :+1: 
Replacing the refresh by press save

File: test/spec/app_spec.rb (hunk 9 above)
Comment (clarissacastella) [reply test/spec/app_spec.rb]: Replacing the refresh by press save
